src/pylero/base_polarion.py

```diff
@@ -127,28 +134,42 @@ class Connection(object):
     POLARION_REPO
     POLARION_USERNAME
     POLARION_PASSWORD
+    POLARION_TOKEN
     POLARION_TIMEOUT
     POLARION_PROJECT
     """
 
     connected = False
     session = None
-    password_retries = 3
+    retries = 3
+    token_enabled = False
 
     @classmethod
     def session(cls):
         if not cls.connected:
             cfg = Configuration()
             # if the password is not supplied in the config file, ask the user
             # for it
-            if not cfg.pwd:
-                cfg.pwd = getpass("Password not in config file.\nEnter Password:")
-            while not cls.connected and cls.password_retries:
+            if cfg.token:
+                cls.token_enabled = True
+            elif cfg.user and not cfg.pwd:
+                cfg.pwd = getpass(
+                    "User name is provided but password not in config file.\n"
+                    + "Note: If want to use personal token please provide token in config file.\n"
+                    + "Enter Password:"
+                )
+            elif not cfg.user or not cfg.token:
+                cfg.token = getpass(
+                    "User name is not provided and access token is also not in config file.\nEnter Token:"
```

Comment:
this pops up when providing user and password but not token, actually with the token you don't need the user at all, so the check here should include the password

Reply:
Fixed